cognite/client/_api/raw.py

```diff
@@ -148,44 +152,50 @@ class RawTablesAPI(APIClient):
     _RESOURCE_PATH = "/raw/dbs/{}/tables"
 
     def __call__(
-        self, db_name: str, chunk_size: Optional[int] = None, limit: Optional[int] = None
-    ) -> Union[Iterator[Table], Iterator[TableList]]:
+        self, db_name: str, chunk_size: int | None = None, limit: int | None = None
+    ) -> Iterator[Table] | Iterator[TableList]:
         """Iterate over tables
 
         Fetches tables as they are iterated over, so you keep a limited number of tables in memory.
 
         Args:
             db_name (str): Name of the database to iterate over tables for
-            chunk_size (int, optional): Number of tables to return in each chunk. Defaults to yielding one table a time.
-            limit (int, optional): Maximum number of tables to return. Defaults to return all items.
+            chunk_size (int | None): Number of tables to return in each chunk. Defaults to yielding one table a time.
+            limit (int | None): Maximum number of tables to return. Defaults to return all items.
+
+        Returns:
+            Iterator[Table] | Iterator[TableList]: No description.
         """
-        for tb in self._list_generator(
+        table_iterator = self._list_generator(
             list_cls=TableList,
             resource_cls=Table,
             resource_path=utils._auxiliary.interpolate_and_url_encode(self._RESOURCE_PATH, db_name),
             chunk_size=chunk_size,
             method="GET",
             limit=limit,
-        ):
-            yield self._set_db_name_on_tables(tb, db_name)
+        )
+        return cast(
+            Union[Iterator[Table], Iterator[TableList]],
+            (self._set_db_name_on_tables(tbl, db_name) for tbl in table_iterator),
+        )
```

Comment:
Correct approach here would be to overload _set_db_name_on_toubles

Reply:
<Moved conversation to slack>